bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableDataSettingsFactory.java

```diff
@@ -0,0 +1,310 @@
+   */
+  private static void buildReadRows(Builder builder, BigtableOptions options) {
+    // TODO: set readPartialRowTimeout for watchdog timer, taken BigtableSession#setupWatchdog()
+    if(options.getRetryOptions().enableRetries()) {
+      builder.readRowsSettings()
+          .setIdleTimeout(ofMillis(options.getRetryOptions().getReadPartialRowTimeoutMillis()))
+          .setRetryableCodes(DEFAULT_RETRY_CODES)
+          .setRetrySettings(defaultRetrySettings(options));
+    } else {
+      builder.readRowsSettings()
+          .setIdleTimeout(ofMillis(options.getRetryOptions().getReadPartialRowTimeoutMillis()))
+          .setSimpleTimeoutNoRetries(DEFAULT_RPC_TIMEOUT)
+          .getRetryableCodes().clear();
+    }
+  }
+
+  /**
+   * This method builds ReadModifyWrite with no retry configurations
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param bulkMutation a {@link BulkOptions} object.
+   */
+  private static void buildReadModifyWrite(Builder builder, BigtableOptions options) {
+    builder.readModifyWriteRowSettings()
+        .setSimpleTimeoutNoRetries(DEFAULT_RPC_TIMEOUT)
+        .getRetryableCodes().clear();
+  }
+
+  /**
+   * This method builds CheckAndMutateRow with no retry configurations
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param bulkMutation a {@link BulkOptions} object.
+   */
+  private static void buildCheckAndMutateRow(Builder builder, BigtableOptions options) {
+    builder.checkAndMutateRowSettings()
+        .setSimpleTimeoutNoRetries(DEFAULT_RPC_TIMEOUT)
+        .getRetryableCodes().clear();
+  }
+
+  /**
+   * To create default RetrySettings, for BigtableDataSettings.
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param bulkMutation a {@link BulkOptions} object.
+   */
+  private static RetrySettings defaultRetrySettings(BigtableOptions options) {
+    RetryOptions retryOptions = options.getRetryOptions();
+    if (!retryOptions.enableRetries()) {
+      // TODO: limiting the retries by adding this makes sense?
+      return RetrySettings.newBuilder()
+          .setMaxAttempts(1).build();
+    }
+
+    // TODO: Make sure other default values are compatible with these configuration.
+    RetrySettings.Builder retryBuilder = RetrySettings.newBuilder()
+        .setInitialRetryDelay(ofMillis(retryOptions.getInitialBackoffMillis()))
+        .setRetryDelayMultiplier(retryOptions.getBackoffMultiplier())
+        .setMaxRetryDelay(ofMillis(retryOptions.getMaxElapsedBackoffMillis()));
+
+    // TODO: verify if maxScanTimeout would cover for maxAttempts or not.
+    retryBuilder.setMaxAttempts(retryOptions.getMaxScanTimeoutRetries());
+
+    // configurations for RPC timeouts
+    retryBuilder
+        .setInitialRpcTimeout(ofMillis(options.getCallOptionsConfig().getShortRpcTimeoutMs()))
+        .setMaxRpcTimeout(ofMillis(retryOptions.getReadPartialRowTimeoutMillis()))
```

Comment:
@igorbernstein2 
I set the `rpcTimeout` with `readPartialTimoutMillis`.. As we are expecting clients must have retry enabled then, would this be fine?

Reply:
This is mixing up different concepts. For readRows you want to set
initialRpcTimeout = maxRpcTimeout = readPartialRowTimeoutMillis

For non-streaming methods you want to set 
initialRpcTimeout = maxRpcTimeout = shortRpcTimeout

I'm guessing the easiest way to get there is have readRows not use this method